packages/magento-product/components/ProductFiltersPro/ProductFiltersPro.tsx

```diff
@@ -50,11 +51,41 @@ export function ProductFiltersPro(props: FilterFormProviderProps) {
   const defaultValues = useMemoObject(toFilterParams(params))
   const form = useForm<ProductFilterParams>({ defaultValues, ...formProps })
 
-  const push = useProductListLinkReplace({ scroll: false })
+  const matchMedia = useMatchMedia()
+  const router = useRouter()
+
+  const push = useProductListLinkReplace()
+
+  // compensate for sticky header on mobile screens when scrolling to '#products'
+  useEffect(() => {
+    const handleRouteChange = (url) => {
+      if (matchMedia.down('md') && url.includes('#products')) {
+        requestAnimationFrame(() => {
+          setTimeout(() => {
+            window.scrollBy(
+              0,
+              (document.querySelector('.LayoutHeader-root')?.getBoundingClientRect().height ?? 0) *
+                -1,
+            )
+          })
+        })
+      }
+    }
```

Comment:
This causes a layout jump if it renders inbetween. Investigate if this works: https://stackoverflow.com/questions/72581132/how-does-the-css-property-scroll-margin-top-and-scroll-padding-top-really-wo